src/goto-programs/remove_returns.cpp

```diff
@@ -46,74 +51,91 @@ class remove_returnst
   void undo_function_calls(
     goto_functionst &goto_functions,
     goto_programt &goto_program);
+
+  symbol_exprt get_or_create_return_value_symbol(const irep_idt &function_id);
 };
 
+symbol_exprt remove_returnst::get_or_create_return_value_symbol(
+  const irep_idt &function_id)
+{
+  const irep_idt symbol_name = id2string(function_id) + RETURN_VALUE_SUFFIX;
+  const symbolt *existing_symbol = symbol_table.lookup(symbol_name);
+  if(existing_symbol != nullptr)
+    return existing_symbol->symbol_expr();
+
+  const symbolt &function_symbol = symbol_table.lookup_ref(function_id);
+  const typet &return_type = to_code_type(function_symbol.type).return_type();
+
+  if(return_type == empty_typet())
+    return symbol_exprt();
+
+  auxiliary_symbolt new_symbol;
+  new_symbol.is_static_lifetime = true;
+  new_symbol.module = function_symbol.module;
+  new_symbol.base_name =
+    id2string(function_symbol.base_name) + RETURN_VALUE_SUFFIX;
+  new_symbol.name = symbol_name;
+  new_symbol.mode = function_symbol.mode;
+  // If we're creating this for the first time, the target function cannot have
+  // been remove_return'd yet, so this will still be the "true" return type:
+  new_symbol.type = return_type;
+  // Return-value symbols will always be written before they are read, so there
+  // is no need for __CPROVER_initialize to do anything:
+  new_symbol.type.set(ID_C_no_initialization_required, true);
+
+  symbol_table.add(new_symbol);
+  return new_symbol.symbol_expr();
+}
+
 /// turns 'return x' into an assignment to fkt#return_value
+/// \param function_id: name of the function to transform
+/// \param function: function to transform
 void remove_returnst::replace_returns(
-  goto_functionst::function_mapt::iterator f_it)
+  const irep_idt &function_id,
+  goto_functionst::goto_functiont &function)
 {
-  typet return_type=f_it->second.type.return_type();
-
-  const irep_idt function_id=f_it->first;
+  typet return_type=function.type.return_type();
```

Comment:
I'd suggest following `clang-format`'s suggestions throughout this patch as at least some of the changes do use the proposed formatting.